Updated Calypso PR template with a checklist (#64814)
Automattic/wp-calypso: .github/PULL_REQUEST_TEMPLATE.md

the linked issue.
-->

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in [Simple](https://wp.me/P9HQHe-k8-p2), [Atomic](https://wp.me/P9HQHe-jW-p2), and [self-hosted Jetpack sites](https://wp.me/PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [ ] Have we sent any new strings [for translation](https://wp.me/PCYsg-1vr-p2) ASAP?

> 

Reviewer: ```suggestion
```

-->

Related to #

#### Things to check

> Here are some important things to check on each PR as an author and reviewer.

Reviewer 2: What do you think of making the language here even more direct about timing and responsibilities?

Something like...

```markdown
#### Pre-merge Checklist

Complete all of the items on this checklist **before** merging the PR into trunk. Both the PR author and review are responsible for ensuring the checklist is completed. Not all items will apply to every PR, and any inapplicable items can just be marked as complete.
```

- [ ] Have you written new tests for your changes?
- [ ] Have you tested the feature in Simple, Atomic, and self-hosted Jetpack sites?
- [ ] Are there any TypeScript or console errors?
- [ ] Are we memoizing when appropriate (for expensive computations)?

> - [ ] Have we sent any new strings for translation ASAP?

Reviewer 2: What do you think about adding links to these, so less familiar people can quickly learn what they need to do?

For example, for the testing one, we could link here: https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md

Author: Good idea @dpasque. Could you please [check these links](https://github.com/Automattic/wp-calypso/pull/64814/commits/5f45a551c9a3e58912d7a5bb6466dae9adf68665) and see if you have any additional?

-->

> Related to #

Reviewer: Could we keep this at the bottom? It's sort of standard to have it at the end across all our repos.

> #### Things to check
> 
> Complete all of the items on this checklist **before** merging the PR into the trunk. Both the PR author and review are responsible for ensuring the checklist is completed. 
> 
> Not all items will apply to every PR, and any inapplicable items can just be marked as complete.

Reviewer: I like where this is going but think it could be more direct. Would also prefer if we left things unchecked instead of checking them off regardless - easier to convey what was done or not done at a glance.

```suggestion
#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
```

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?

> - [ ] Have you tested the feature in [Simple](https://atomicuniversity.wordpress.com/what-is-atomic/make-a-site-atomic/simple-vs-atomic), [Atomic](https://atomicuniversity.wordpress.com/what-is-atomic), and [self-hosted Jetpack sites](https://fieldguide.automattic.com/jurassic-ninja)?

Reviewer: It doesn't disclose much but we should use short links here anyway. `h\ttps://wp.me/P9HQHe-k8` etc

Would be good to replicate some of this documentation in `wp-calypso/docs` so we can keep all these links pointing to something public. (We have a few open source contributions that come in here and there)

> - [ ] Are there any TypeScript or console errors?

Reviewer: ```suggestion
- [ ] Have you checked for TypeScript, React or other console errors?
```
Thinking "Are there any errors" is potentially open ended. Asking if you've checked is easier to give a yes/no checkbox answer to.

> - [ ] Are we memoizing when appropriate (for expensive computations)?

Reviewer: It's not really clear when this is appropriate, or how I can identify an expensive computation that needs it.

Is there anything in the docs we could link to here? Only mention I found was [this page on data.](https://github.com/Automattic/wp-calypso/blob/65497ddd515ad181da7c486ecfad95a4fbf08db7/docs/our-approach-to-data.md) Should we add something?

Author: Found some extra links and added them to the docs. Thanks!

> - [ ] Have we sent any new strings [for translation](https://fieldguide.automattic.com/i18n/how-translation-projects-work) ASAP?

Reviewer: Short link needed here too.

@Automattic/i18n do we have any documentation we could move into calypso's docs folder and link to there instead?

Reviewer 3: Since it's for Calypso, explaining just the usage of `[Status] String Freeze` might be the best solution. The original announcement post (p4TIVU-5Jq-p2) has a pretty good description of the process, so maybe it could be added to calypso docs, with some small updates/corrections?
